community/server/src/main/java/org/neo4j/server/database/CypherExecutor.java

```diff
@@ -74,8 +80,24 @@ public void stop() throws Throwable
 
     public QuerySession createSession( HttpServletRequest request )
     {
-        InternalTransaction transaction = service.beginTransaction( KernelTransaction.Type.implicit, AccessMode.Static.FULL );
+        InternalTransaction transaction = getInternalTransaction( request );
         TransactionalContext context = new Neo4jTransactionalContext( service, transaction, txBridge.get(), locker );
         return new ServerQuerySession( request, context );
     }
+
+    private InternalTransaction getInternalTransaction( HttpServletRequest request )
+    {
+        long customTimeout = getTransactionTimeout( request, log );
+        return customTimeout > 0 ? beginCustomTransaction( customTimeout ) : beginDefaultTransaction();
```

Comment:
Isn't this an unnecessary check (change overall)? timeout: 0 means no timeout, doesn't it?